ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/query/BlockchainQueries.java

```diff
@@ -624,22 +624,26 @@ public Optional<TransactionReceiptWithMetadata> transactionReceiptByTransactionH
     // getTransactionLocation should not return if the TX or block doesn't exist, so throwing
     // on a missing optional is appropriate.
     final TransactionLocation location = maybeLocation.get();
-    final Block block = blockchain.getBlockByHash(location.getBlockHash()).orElseThrow();
-    final Transaction transaction =
-        block.getBody().getTransactions().get(location.getTransactionIndex());
-
     final Hash blockhash = location.getBlockHash();
+    final int transactionIndex = location.getTransactionIndex();
+
+    final Block block = blockchain.getBlockByHash(blockhash).orElseThrow();
+    final Transaction transaction = block.getBody().getTransactions().get(transactionIndex);
+
     final BlockHeader header = block.getHeader();
     final List<TransactionReceipt> transactionReceipts =
         blockchain.getTxReceipts(blockhash).orElseThrow();
-    final TransactionReceipt transactionReceipt =
-        transactionReceipts.get(location.getTransactionIndex());
+    final TransactionReceipt transactionReceipt = transactionReceipts.get(transactionIndex);
 
     long gasUsed = transactionReceipt.getCumulativeGasUsed();
-    if (location.getTransactionIndex() > 0) {
-      gasUsed =
-          gasUsed
-              - transactionReceipts.get(location.getTransactionIndex() - 1).getCumulativeGasUsed();
+    int logIndexOffset = 0;
+    if (transactionIndex > 0) {
+      gasUsed -= transactionReceipts.get(transactionIndex - 1).getCumulativeGasUsed();
+      logIndexOffset =
+          IntStream.range(0, transactionIndex)
+              .parallel()
+              .map(i -> transactionReceipts.get(i).getLogs().size())
+              .sum();
```

Comment:
since usually we only have few hundreds txs in a block, have you any number to confirm that doing the sum in parallel is faster than doing it serially?

Reply:
Not really any numbers there. I am just thinking that it does not hurt to map and sum the numbers in parallel.

Reply:
just thinking that a quick JMH test could be enough to spot the difference in performance, between serial/parallel

Reply:
I have no idea how to do a JMH benchmark, but I tried the query with `time curl` and deducting the `user` and `sys` time
```GraphQL
query {
	blocks (from: 18676591 to: 18676640) {
		transactions { gasUsed }
	}
}
```
Every time GraphQL transverses into a transaction it would first obtain the receipt for the amount of gas used, thus running the code for obtaining log indices in the receipt.

So with 8235 transactions in these 50 blocks the results are as follows:
|`.parallel()`|Time averaged over 10 queries|
|-----------------|--------:|
| with | 8.5605 s |
| without | 8.6308 s |

So it may not be worth the hassle to parallelise

Reply:
Ok, indeed for small lists there should not be a big difference.
Just for reference, an example of a JMH test in Besu is [RLPBench](https://github.com/hyperledger/besu/blob/8f5a4443cc2e511d0df6eab5dca2723d104a46d7/ethereum/rlp/src/jmh/java/org/hyperledger/besu/ethereum/rlp/RLPBench.java#L31)